OSDOCS-5782 Z-stream release notes for 4.12.14 (#59110)
openshift/openshift-docs: release_notes/ocp-4-12-release-notes.adoc

[id="ocp-4-12-13-updating"]
==== Updating

To update an existing {product-title} 4.12 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI] for instructions.

[id="ocp-4-12-14"]
=== RHBA-2023:1858 - {product-title} 4.12.14 bug fix update

Issued: 2023-04-24

{product-title} release 4.12.14 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2023:1858[RHBA-2023:1858] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:1857[RHBA-2023:1857] advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.12.14 --pullspecs
----

[id="ocp-4-12-14-features"]
==== Features

[id="ocp-4-12-14-cloud-provider-openstack-1.25"]
===== Cloud provider OpenStack is updated to 1.25

> With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.5. Updates include:

Reviewer: ```suggestion
With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.5.  The update includes the addition of an annotation for real load balancer IP addresses and the global source for `math/rand` packages are seeded in `main.go`. 
```

Author: Changed.

Reviewer 2: I like this too. There should be a comma before the "and" but otherwise looks good!

> * Adds an annotation for real load balancer IP addresses.

Reviewer: I liked @opayne1 recommendation of keeping it simple.

[id="ocp-4-12-13-updating"]
==== Updating

To update an existing {product-title} 4.12 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI] for instructions.

[id="ocp-4-12-14"]
=== RHBA-2023:1858 - {product-title} 4.12.14 bug fix update

Issued: 2023-04-24

{product-title} release 4.12.14 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2023:1858[RHBA-2023:1858] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:1857[RHBA-2023:1857] advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.12.14 --pullspecs
----

[id="ocp-4-12-14-features"]
==== Features

[id="ocp-4-12-14-cloud-provider-openstack-1.25"]
===== Cloud provider OpenStack is updated to 1.25

> With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.5.

Reviewer 2: I am not sure if it was considered or not, but it might be more concise to include the bullet with this sentence. So,
```suggestion
With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.5 which implements support for concurrent atomic routes updates.
```
Either way you want to do it is fine though!

Author: I misunderstood a previous suggestions so I changed the list of bullet points. Can you take another look? @opayne1 Thank you!

[id="ocp-4-12-13-updating"]
==== Updating

To update an existing {product-title} 4.12 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI] for instructions.

[id="ocp-4-12-14"]
=== RHBA-2023:1858 - {product-title} 4.12.14 bug fix update

Issued: 2023-04-24

{product-title} release 4.12.14 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2023:1858[RHBA-2023:1858] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:1857[RHBA-2023:1857] advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.12.14 --pullspecs
----

[id="ocp-4-12-14-features"]
==== Features

[id="ocp-4-12-14-cloud-provider-openstack-1.25"]
===== Cloud provider OpenStack is updated to 1.25

With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.5.

[id="ocp-4-12-14-summary-of-openstack-changes"]

> ====== Overview of Cloud Provider 1.25 updates

Reviewer 2: I am not 100% about a level 6 heading here. I didn't see anything in the style guide against it, so it is up to you, but I almost think it would be clearer to the reader if  you  keep it with the above and just say 

> With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.5. Updates include:
*bullet
*bullet

Author: Change made.

> * Adds annotation for real load balancer IP addresses.

Reviewer 2: ```suggestion
* Adds an annotation for real load balancer IP addresses.
```

Author: Change made.

> * The global source for `math/rand` is seeded in `main.go`.

Reviewer 2: ```suggestion
* The global source for `math/rand` is now seeded in `main.go`.
```

Author: Change made.

[id="ocp-4-12-13-updating"]
==== Updating

To update an existing {product-title} 4.12 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI] for instructions.

[id="ocp-4-12-14"]

> === RHBA-2023:1858 - {product-title} 4.12.14 bug fixes and features

Reviewer: ```suggestion
=== RHBA-2023:1858 - {product-title} 4.12.14 bug fixes 
```

Reviewer: I was keeping things consistent with above and I'm not sure we've ever changed the titles to reflect features.

Author: Change made, thanks!

Issued: 2023-04-24

{product-title} release 4.12.14 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2023:1858[RHBA-2023:1858] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:1857[RHBA-2023:1857] advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.12.14 --pullspecs
----

[id="ocp-4-12-14-features"]
==== Features

[id="ocp-4-12-14-cloud-provider-openstack-1.25"]
===== Cloud provider OpenStack is updated to 1.25

> With this release, Cloud Provider {rh-openstack-first} is updated to 1.25.

Author: Changed to 1.25.5

[id="ocp-4-12-14-summary-of-openstack-changes"]

> ====== Quick summary of changes

Reviewer: ```suggestion
====== Overview of Cloud Provider 1.25 updates
```

Author: Change made.

> ====== Quick summary of changes

Author: Thanks!

* Adds annotation for real load balancer IP addresses.
* The global source for `math/rand` is seeded in `main.go`.

> * Improves performance of the `ListRoutes` method, especially when an {rh-openstack} project contains hundreds of non-Kubernetes nodes.

Reviewer: I guess given @mandre suggestion above that we can omit these.

> * Implements a support for concurrent atomic routes updates.

Author: Changes made, thank you!
